core/internal/tensorboard/emitter.go

```diff
@@ -261,37 +253,42 @@ func (e *tfEmitter) EmitTable(
 	return nil
 }
 
-func (e *tfEmitter) EmitImage(
+func (e *tfEmitter) EmitImages(
 	key pathtree.TreePath,
-	img wbvalue.Image,
+	images []wbvalue.Image,
 ) error {
-	maybeRunFilePath, err := runRelativePath(
-		filepath.Join("media", "images"),
-		fmt.Sprintf(".%s", img.Format),
-	)
-	if err != nil {
-		return err
-	}
-	runRelativePath := *maybeRunFilePath
-	fsPath := filepath.Join(e.settings.GetFilesDir(), string(runRelativePath))
-
-	if _, err := os.Stat(fsPath); !os.IsNotExist(err) {
-		return fmt.Errorf("image file exists: %v", err)
+	// We assume that all images have the same format, height, and width.
+	// This aligns with the legacy way of logging images in TensorBoard.
+	format := images[0].Format
+	height := images[0].Height
+	width := images[0].Width
+
+	imagePaths := []paths.RelativePath{}
+	for _, img := range images {
+		maybeRunFilePath, err := runRelativePath(
+			filepath.Join("media", "images"),
+			fmt.Sprintf(".%s", img.Format),
+		)
+		if err != nil {
+			return err
+		}
+
+		runRelativePath := *maybeRunFilePath
+		fsPath := filepath.Join(e.settings.GetFilesDir(), string(runRelativePath))
+
+		if err := e.writeDataToPath(fsPath, img.EncodedData); err != nil {
+			return err
+		}
+
+		e.mediaFiles = append(e.mediaFiles, string(runRelativePath))
```

Comment:
Note: on error, we may upload some image files without using them, since `tfEmitter.Emit` will still be called. I think that's fine, so no change necessary, but I did intentionally put state modifications after error handling.